client/src/NewCourseDialog.jsx

```diff
@@ -76,23 +109,29 @@ const NewCourseDialog = (props) => {
     }
 
     return (
+        <>
         <Dialog fullScreen open={open} onClose={stopCreating}>
             <AppBar position="static" component="nav">
                 <Toolbar>
                     <IconButton color="inherit" onClick={stopCreating}>
                         <Icon>close</Icon>
                     </IconButton>
                     <Typography sx={{ flex: 1, paddingLeft: 1 }} variant="h6"> Create New Course for Trimester {activeTri.tri} </Typography>
-                    <Button color="inherit" onClick={createCourse}> Create </Button>
+                    <Button color="inherit" onClick={createCourse} disabled={!formatValid}> Create </Button>
                 </Toolbar>
             </AppBar>
             <Box sx={{padding: 3, margin: "auto", marginTop: 0}}>
                 <Typography variant="h5"> Basic Info </Typography>
                 <Divider sx={{marginBottom: 3}} />
                 <Stack spacing={2}>
-                    <TextField value={courseName} label="Course Name" sx={{ width: 500 }} onChange={(e) => setCourseName(e.target.value)}/>
+                    <TextField value={courseName} label="Course Name" sx={{ width: 500 }} onChange={handleNameChange} error={!nameValid} 
+                        helperText={courseName.length === 0 ? "This field cannot be empty" : courseName.length > 50 ? "This field  is too long" : ""} 
+                    />
                     <Box>
-                        <TextField value={courseCode} label="Course Code" sx={{ width: 200 }} onChange={(e) => setCourseCode(e.target.value)} />
+                        <TextField value={courseCode} label="Course Code" sx={{ width: 200 }} onChange={handleCodeChange} 
+                            error={!codeValid} 
+                            helperText={!codeValid ? "Invalid course code" : ""} 
+                        />
                         <FormControlLabel control={<Checkbox defaultChecked />} label="Course Info Incomplete" sx={{padding: 0.7, paddingLeft: 6}}/>
```

Comment:
I think this should be unchecked by default

Reply:
Hi Chris,

I can normalize the course codes on my end before sending them over to you. 

In regards to the other questions about the "Course Info Incomplete" checkbox, it is a placeholder with no functionality for the time being, it's there for in the future when we implement course template editing or allowing users to see a template is not fully done/accurate. 

The toast popup issue is odd and I do not recall running into it when testing it this time, but I will do further testing and get back to you if I need you to replicate it.

As for everything starting red I do agree with you but I have to make a UX decision that doesn't confuse the user and remains consistent so I've decided I'll make another issue for it later (I figured UX changes like that can be delayed until later when we have an actual website design)

Reply:
> I can normalize the course codes on my end before sending them over to you.

Sounds good, thanks.

> I figured UX changes like that can be delayed until later when we have an actual website design.

I agree it can be delayed as mentioned in my first comment.

> The toast popup issue is odd and I do not recall running into it when testing it this time, but I will do further testing and get back to you if I need you to replicate it.

It's not a huge issue, it just caught my eye and confused me for a second. I did check that it didn't add the course and it didn't so it doesn't affect the process at all.

> In regards to the other questions about the "Course Info Incomplete" checkbox, it is a placeholder with no functionality for the time being.

Ah ok. That answers that question then. 

---

I think that leaves two changes to be made then. 

- Normalising course codes
- Fixing toast colour change